extensions/smallrye-health/runtime/src/main/java/io/quarkus/smallrye/health/runtime/SmallRyeHealthHandlerBase.java

```diff
@@ -41,19 +45,43 @@ private void doHandle(RoutingContext ctx) {
             Arc.container().instance(CurrentIdentityAssociation.class).get().setIdentity(user.getSecurityIdentity());
         }
         SmallRyeHealthReporter reporter = Arc.container().instance(SmallRyeHealthReporter.class).get();
-        SmallRyeHealth health = getHealth(reporter, ctx);
-        HttpServerResponse resp = ctx.response();
-        if (health.isDown()) {
-            resp.setStatusCode(503);
+        getHealth(reporter, ctx).subscribe().with(health -> {
+            HttpServerResponse resp = ctx.response();
+            if (health.isDown()) {
+                resp.setStatusCode(503);
+            }
+            resp.headers().set(HttpHeaders.CONTENT_TYPE, "application/json; charset=UTF-8");
+            try (ChunkedResponseOutputStream outputStream = new ChunkedResponseOutputStream(resp)) {
+                reporter.reportHealth(outputStream, health);
+            } catch (Throwable e) {
+                log.error("Unable to prepare health response message", e);
+            }
+            resp.end((Handler<AsyncResult<Void>>) null);
```

Comment:
why is "end()" not enough? Passing null is rather dangerous.

Reply:
Calling `end()` would internally create a `Promise` and return `Future` no-one would be interested in.

So this code is more efficient. I can change this in the next commit.